src/helpers/checks.jl

```diff
@@ -130,38 +201,41 @@ For this we require either a second-order retraction or a critical point ``p`` o
-behaves in error ``\mthcal O(t^2)``, which indicates that the gradient is correct,
-cf. also Section 4.8 [^Boumal2022].
+or in other words, that the error between the function ``f`` and its second order Taylor
+behaves in error ``\mathcal O(t^3)``, which indicates that the gradient is correct,
+cf. also Section 6.8 [^Boumal2022].
+
+Note that if the errors are below the given tolerance and the method is exact,
+no plot will be generated,
 
 # Keyword arguments
 
-* `check_gradient` – (`true`) check whether ``\operatorname{grad} f(p) \in T_p\mathcal M``.
-* `check_vector` – (`true`) check whether ``\operatorname{Hess} f(p)[X]``  \in T_p\mathcal M`` using `is_vector`.
-* `check_symmetry` – (`true`) check whether TODO
-* `check_linearity` – (`true`) check whether TODO
-* `check_mode` (`:Default`) specify the mode, by default we assume to have a second order retraction given by `retraction_method=`
+* `check_gradient`   – (`true`) check whether ``\operatorname{grad} f(p) \in T_p\mathcal M``.
+* `check_linearity`  – (`true`) check whether the Hessian is linear, see [`is_Hessian_linear`](@ref) using `a`, `b`, `X`, and `Y`
+* `check_symmetry`   – (`true`) check whether the Hessian is symmetric, see [`is_Hessian_symmetric`](@ref)
+* `check_vector`     – (`true`) check whether ``\operatorname{Hess} f(p)[X]``  \in T_p\mathcal M`` using `is_vector`.
+* `mode`             - (`:Default`) specify the mode, by default we assume to have a second order retraction given by `retraction_method=`
   you can also this method if you already _have_ a cirtical point `p`.
   Set to `:CritalPoint` to use [`gradient_descent`](@ref) to find a critical point.
   Note: This requires (and evaluates) new tangent vectors `X` and `Y`
 
-* `a`, `b` – two real values to check linearity of the Hessian (if `check_linearity=true`)
-* `N` (`101`) – number of points to check within the `log_range` default range ``[10^{-8},10^{0}]``
-* `io` – (`nothing`) provide an `IO` to print the check result to
+* `a`, `b`            – two real values to check linearity of the Hessian (if `check_linearity=true`)
+* `N`                 - (`101`) number of points to check within the `log_range` default range ``[10^{-8},10^{0}]``
+* `eaxctness_tol`     - (`1e-12`) is all errors are below this tolerance, the check is considered to be exact
+* `io`                – (`nothing`) provide an `IO` to print the check result to
 * `gradient`          - (`grad_f(M, p)`) instead of the gradient _function_ you can also provide the gradient at `p` directly
-* `Hessian`           - (`Hess_f(M,p)`) instead of the Hessian _function_ you can provide the result of `\operatorname{Hess} f(p)[X]` directly.
-  Note that evaluations of the Hessian might still be necessary for checking linearity and symmetry.
+* `Hessian`           - (`Hess_f(M,p)`) instead of the Hessian _function_ you can provide the result of ``\operatorname{Hess} f(p)[X]``` directly.
```

Comment:
```suggestion
* `Hessian`           - (`Hess_f(M, p, X)`) instead of the Hessian _function_ you can provide the result of ``\operatorname{Hess} f(p)[X]`` directly.
```